compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -5724,6 +5724,37 @@ object Types {
             foldOver(xs, tp)
   end NamedPartsAccumulator
 
+  class ExistsLocalRef(using Context) extends TypeAccumulator[Boolean]:
```

Comment:
Does this actually make type avoidance faster? It's not clear to me because this still requires traversing the type to see if there's something to avoid.

Reply:
We don't have much empirical evidence. It avoids computing `localSyms` and it has shortcut for references to type or term parameters. Or we can enhance the `TypeOps.avoid`. This part can come later if there indeed a noticeable speedup. For now, I'll remove it from the PR.